forui/lib/src/widgets/tooltip.dart

```diff
@@ -0,0 +1,391 @@
+
+  @override
+  State<FTooltip> createState() => _FTooltipState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('controller', controller))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('tipAnchor', tipAnchor))
+      ..add(DiagnosticsProperty('childAnchor', childAnchor))
+      ..add(DiagnosticsProperty('shift', shift))
+      ..add(DiagnosticsProperty('manual', manual))
+      ..add(DiagnosticsProperty('hoverEnterDuration', hoverEnterDuration))
+      ..add(DiagnosticsProperty('hoverExitDuration', hoverExitDuration))
+      ..add(DiagnosticsProperty('pressExitDuration', pressExitDuration))
+      ..add(DiagnosticsProperty('tipBuilder', tipBuilder));
+  }
+}
+
+class _FTooltipState extends State<FTooltip> with SingleTickerProviderStateMixin {
+  late FTooltipController _controller;
+  Key _fencingToken = UniqueKey();
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = widget.controller ?? FTooltipController(vsync: this);
+  }
+
+  @override
+  void didUpdateWidget(covariant FTooltip old) {
+    super.didUpdateWidget(old);
+    if (widget.controller != old.controller) {
+      if (old.controller == null) {
+        _controller.dispose();
+      }
+
+      _controller = widget.controller ?? FTooltipController(vsync: this);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final style = widget.style ?? context.theme.tooltipStyle;
+
+    var child = widget.child;
+    if (!widget.manual) {
+      child = CallbackShortcuts(
+        bindings: {
+          LogicalKeySet(LogicalKeyboardKey.escape): _exit,
+        },
+        child: Focus(
+          onFocusChange: (focused) async => focused ? _enter() : _exit(),
+          child: child,
+        ),
+      );
+
+      // TODO: haptic feedback.
+      if (touchPlatforms.contains(defaultTargetPlatform)) {
+        child = GestureDetector(
+          onLongPressStart: (_) async {
+            _fencingToken = UniqueKey();
+            await _controller.show();
+          },
+          onLongPressEnd: (_) async {
+            final fencingToken = _fencingToken = UniqueKey();
+            await Future.delayed(widget.pressExitDuration);
+
+            if (fencingToken == _fencingToken) {
+              await _controller.hide();
+            }
+          },
+          child: child,
+        );
+      } else {
+        child = MouseRegion(
+          onEnter: (_) => _enter(),
+          onExit: (_) => _exit(),
+          // We have to use a Listener as GestureDetector's arena implementation allows only 1 gesture to win. It is
+          // problematic if the child is a button. See https://github.com/flutter/flutter/issues/92103.
+          child: Listener(
+            onPointerDown: (_) => _exit(),
+            child: child,
+          ),
+        );
+      }
+    }
```

Comment:
Is there a point checking for platform? It could probably make more sense for long tap + hover to activate the tooltip. That way the usage is consistent and we don't have to deal with awkward use cases where eg. touchscreen windows laptop.

Reply:
I don’t have any strong opinion on it, I just thought that it might be weird if you could long press to view the tooltip on desktop.